test/safe_test.rb

```diff
@@ -128,4 +128,41 @@ def test_FakeFS_method_deactivates_FakeFS_when_block_raises_exception
 
     refute FakeFS.activated?
   end
+
+  def test_FakeFS_activate_method_stubs_File_class_but_not_IO
+    old_file = ::File
+    old_io = ::IO
+
+    ::FakeFS.activate!
+
+    refute_equal old_file, ::File
+    assert_equal old_io, ::IO
+  ensure
+    ::FakeFS.deactivate!
+  end
+
+  def test_FakeFS_activate_method_stubs_IO_class_if_explicity_asked
+    old_file = ::File
+    old_io = ::IO
+
+    ::FakeFS.activate!(io_mocks: true)
+
+    refute_equal old_file, ::File
+    refute_equal old_io, ::IO
+  ensure
+    ::FakeFS.deactivate!
```

Comment:
this would be a nicer pattern :)
```
::FakeFS.activate!(io_mocks: true) do
  ... stuff ...
end
```

Reply:
... yeah that's another rabbithole where `.with` can't distinguish between these 2 states

Reply:
Yep I gave up on that part :D